### ISSUE TITLE
MSLEVEL=2 empty in mgf

### ISSUE THREAD

**Issue by mbarthele** [mzmine/mzmine2#516]
### Setup

I am reporting a problem with MZmine 2 version 2.36. 

I've created a mgf file by exporting a peak list (Export for GNPS). From this mgf, it appears that there are some ions which have a MS2 but this MS2 seems to be empty.
(Example:
BEGIN IONS
FEATURE_ID=185
PEPMASS=165.9916
SCANS=185
RTINSECONDS=510.13
CHARGE=1
MSLEVEL=2
END IONS
)

But in my peak list, if I look to the MS2 of this ion, fragments exist with a very good intensity. Moreover, the Export MS/MS Pattern of this ion gives the fragment list correctly
(Example:
93.07044982910156 481637.0
111.08090209960938 228057.0
121.06498718261719 245091.296875
125.05974578857422 155582.0625
125.09626007080078 414801.46875
139.03916931152344 143278.609375
139.07537841796875 710745.1875
149.05978393554688 446169.03125
153.09100341796875 631558.125
167.07046508789062 585678.5
181.08583068847656 203007.515625
209.08084106445312 2371511.5
227.09210205078125 465902.0625
)

I don't understand where the problem is occuring and why some MS2 are missing. They exist in the peak list but now in the mgf after exportation.

Thank you very much,
Regards,
Morgane Barthélemy

**Comment by tomas-pluskal** [comment]
Please provide your .mzmine project file (e.g., via Dropbox).

**Comment by mbarthele** [comment]
Thank you for your answer
You can find below my project file with only the final aligned peak list. Let me know if you need the previous steps of the project (deconvolution ...)

https://www.dropbox.com/s/1i7po0b3knoagwp/M%20BARTHE%20project.mzmine?dl=0

Best regards,
Morgane BArthélemy

**Comment by tomas-pluskal** [comment]
And how exactly did you perform the export?

**Comment by mbarthele** [comment]
I used the option export for GNPS

**Comment by tomas-pluskal** [comment]
Yeah you mentioned that already. I am asking about all the parameter values.

**Comment by mbarthele** [comment]
- Centroid mass detector with the noise level set to 1.0E5 for MS level set to all
- ADAP chromatogram builder (minimum group size of scans of 5, minimum group intensity threshold of 1.0E5, minimum highest intensity of 1.0E5 and m/z tolerance of 0.002 or 5 ppm)
- Wavelets (ADAP) algorithm for deconvolution with S/N threshold of 10, intensity window SN, minimum feature height of 1000, coefficient area threshold of 100, peak duration range between 0.01 and 0.5 and the RT wavelet range between 0.001 and 0.05 (m/z and RT range for MS2 scan pairing was set to 0.001 Da and 0.05 min respectively)
- Isotopic peaks grouper algorithm with a m/z tolerance of 0.003 (5 ppm), a RT tolerance of 0.1 (absolute), a maximum charge of 2 and the representative isotope used was the most intense
- Join aligner method: m/z tolerance of 0.001 or 5.0 ppm, weight for m/z of 0.001, RT tolerance of 0.3 min, weight for RT of 0.1
- Gap-filled with the peak finder module: intensity tolerance of 90%, m/z tolerance of 0.001 or 5.0 ppm and RT tolerance of 0.1 min.
- Filtered for MS2 scans only

**Comment by lfnothias** [comment]
Are these QTOF or QE data ? 
In any case, 1E5 for mass detection on level 2 would be way too high for the QTOF and for the QE  … Just use 0 if you are not sure about it.
Moreover, the mass detection level used seems inconsistent the chromatogram builder / deconvolution parameters used.

LF

**Comment by tomas-pluskal** [comment]
The results are consistent with the noise level you set during the mass detection step. But I guess the GNPS export module should be updated to avoid exporting empty scans.

**Comment by robinschmid** [comment]
Agreed. I will change that in a future version.
